medical-system/src/components/shared/Table.tsx

```diff
@@ -0,0 +1,128 @@
+import { ReactNode, useEffect, useState } from "react";
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { collection, getDocs } from "firebase/firestore";
+import { db } from "../../auth/firebase-config";
+
+export type User = {
+  name: string;
+  email: string;
+  mobile: string;
+  address: string;
+  dateBirth: string;
+  delete?: ReactNode;
+  edit?: ReactNode;
+};
+
+const columns = [
+  {
+    header: "Name",
+    accessorKey: "name",
+  },
+  {
+    header: "Email",
+    accessorKey: "email",
+  },
+  {
+    header: "Mobile",
+    accessorKey: "mobile",
+  },
+  {
+    header: "Address",
+    accessorKey: "address",
+  },
+  {
+    header: "Date Birth",
+    accessorKey: "dateBirth",
+  },
+];
```

Comment:
to by można było przenieść do pliku obok tego komponentu np useColumns, i w nim stworzyć hooka który zwróci nam kolumny i następnie go wywołać w komponencie const { columns } = useColumns();

Poza tym, brakuje nam tutaj dodanego typu, przez co nie wiemy jakie własciwosci obiektu dla kolumn mogą być ustalone, szkoda